### PR TITLE
Generate version documentation from single source of truth

### PR DESCRIPTION
**Pull request by potiuk**
We used to maintain supported versions separately in the docs
and it led to discrepancies. Now we have single source of truth which
is used to generate it automatically with pre-commits

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by eladkal — APPROVED** [review]
Nice!
Should we mention it in
https://github.com/apache/airflow/blob/main/dev/README_RELEASE_AIRFLOW.md#update-main-with-latest-release-details

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
> Nice! Should we mention it in https://github.com/apache/airflow/blob/main/dev/README_RELEASE_AIRFLOW.md#update-main-with-latest-release-details

Since pre-commit will fail without this change, I think it should be fine

File: scripts/ci/pre_commit/supported_versions.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from pathlib import Path
+
+from tabulate import tabulate
+
+AIRFLOW_SOURCES = Path(__file__).resolve().parent.parent.parent.parent
```
Comment (kaxil) [inline scripts/ci/pre_commit/supported_versions.py:22]: 😄 `.parent.parent.parent.parent`

File: scripts/ci/pre_commit/supported_versions.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+SUPPORTED_VERSIONS = (
+    ("2", "2.2.3", "Supported", "Dec 17, 2020", "TBD", "TBD"),
+    ("1.10", "1.10.15", "EOL", "Aug 27, 2018", "Dec 17, 2020", "June 17, 2021"),
+    ("1.9", "1.9.0", "EOL", "Jan 03, 2018", "Aug 27, 2018", "Aug 27, 2018"),
+    ("1.8", "1.8.2", "EOL", "Mar 19, 2017", "Jan 03, 2018", "Jan 03, 2018"),
+    ("1.7", "1.7.1.2", "EOL", "Mar 28, 2016", "Mar 19, 2017", "Mar 19, 2017"),
+)
+
+
+def replaceTextBetween(file: Path, start: str, end: str, replacement_text: str):
```
Comment (kaxil) [inline scripts/ci/pre_commit/supported_versions.py]:
```suggestion
def replace_text_between(file: Path, start: str, end: str, replacement_text: str):
```

nit ! -- don't mind if you don't change it

Comment (potiuk) [reply scripts/ci/pre_commit/supported_versions.py]: ah. yeah :)

File: scripts/ci/pre_commit/supported_versions.py (hunk 1 above)
Comment (potiuk) [reply scripts/ci/pre_commit/supported_versions.py:22]: grand-grand-grand-parent

**Comment by potiuk** [comment]
> Nice!
> Should we mention it in

good point

**Comment by potiuk** [comment]
Updated @eladkal @kaxil

**Review by kaxil — APPROVED** [review]

**Pull request merged**